### PR TITLE
Docs(web-react, web-twig): Add Dropdown demo #DS-895

### PR DESCRIPTION
**Pull request by pavelklibani**
## Description

All demos in **web-react**, **web-twig** and **web** are same.

### Issue reference

[Demo: Dropdown má ve všech technologiích stejná dema](https://jira.lmc.cz/browse/DS-895)

### CODE REVIEW

File: packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const dropdownTrigger = ({ trigger: { className, ref, ...restOf } }: DropdownRenderProps) => (
+    <Button UNSAFE_className={className} ref={ref as Ref<HTMLButtonElement>} {...restOf}>
+      Button as anchor
+    </Button>
+  );
+
+  const dropdownContent = (
+    <>
+      <a href="#info" className="d-flex mb-400">
+        <Icon name="info" />
```
Comment (literat) [inline packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx:18]: Yeah, we definitely need an Icon With Text component :-/

File: packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { Dropdown } from '..';
+
+const DropdownDefault = () => {
+  const dropdownTrigger = ({ trigger: { className, ref, ...restOf } }: DropdownRenderProps) => (
+    <Button UNSAFE_className={className} ref={ref as Ref<HTMLButtonElement>} {...restOf}>
+      Button as anchor
+    </Button>
+  );
+
+  const dropdownContent = (
```
Comment (literat) [inline packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx:15]: What about some `DropdownContentFactory` component? I think this content is repeated, isn't it?

**Comment by literat** [comment]
Btw, the space in dropdown content is underlined on hover. But in the web demo, it is not.

Screenshot 2023-09-01 at 13 13 43

**Review by literat — CHANGES_REQUESTED** [review]
Underlined space on hover. Otherwise, it is looking good.

File: packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <>
+      <a href="#info" className="d-flex mb-400">
+        <Icon name="info" />
+        &nbsp;<Text UNSAFE_className="ml-300">Information</Text>
+      </a>
+      <a href="#link" className="d-flex mb-400">
+        <Icon name="link" />
+        &nbsp;
+        <Text UNSAFE_className="ml-300">Bibendum aliquam, fusce integer sit amet congue non nulla aliquet enim</Text>
```
Comment (crishpeen) [inline packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx:24]: This creates a <p> inside the <a>. Which shouldn't be.
Can we implement it the same way it is in twig?

```
<a href="#" class="d-flex mb-400">
            <Icon name="info" UNSAFE_className="mr-400" />
            <span>Information</span>
            ...
```

File: packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx (hunk 2 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx]: good point, created new `<DropdownContentFactory />` component

**Review by literat — APPROVED** [review]
LGTM 👍 Good Job!

File: packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { Ref } from 'react';
+import { DropdownRenderProps } from '../../../types';
+import { Button } from '../../Button';
+import { Dropdown } from '..';
+import DropdownContentFactory from './DropdownContentFactory';
+
+const dropdownContent = [
```
Comment (crishpeen) [inline packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx:7]: as this is the same everywhere, couldn't we have it only in one place?

Comment (pavelklibani) [reply packages/web-react/src/components/Dropdown/demo/DropdownDefault.tsx:7]: good point, moved it to separate file 👍

**Pull request merged**